test/paymentmethods.collective.to.collective.test.js

```diff
@@ -0,0 +1,438 @@
+      };
+      await models.Transaction.createMany(transactions, transactionsDefaultValue);
+      await models.Transaction.createManyDoubleEntry(transactions, transactionsDefaultValue);
+    });
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      // And given that the endpoint for creating customers on Stripe
+      // is patched
+      utils.stubStripeCreate(sandbox, { charge: { currency: 'usd', status: 'succeeded' } });
+      // And given the stripe stuff that depends on values in the
+      // order struct is patch. It's here and not on each test because
+      // the `totalAmount' field doesn't change throught the tests.
+      utils.stubStripeBalance(sandbox, ORDER_TOTAL_AMOUNT, 'usd',
+                              0,
+                              STRIPE_FEE_STUBBED_VALUE); // This is the payment processor fee.
+    });
+
+    afterEach(() => sandbox.restore());
+
+
+    it('collective balance for Admin Member matches sum of transactions amounts', async () => {
+      // getting balance of transactions that were just created
+      const reducer = (accumulator, currentValue) => accumulator + currentValue;
+
+      // multiplying by -1 to get the Collective 1 Credit values
+      const balance = (-1) * transactions.map(t => t.netAmountInCollectiveCurrency).reduce(reducer, 0);
+
+      //finding opencollective payment method for Collective 1
+      openCollectivePaymentMethod = await models.PaymentMethod.findOne({ where: {type: 'collective', CollectiveId: collective1.id}});
+
+      // get Balance given the created user
+      const ocPaymentMethodBalance = await openCollectivePaymentMethod.getBalanceForUser(user1);
+
+      expect(balance).to.equal(ocPaymentMethodBalance.amount);
+      expect(collective1.currency).to.equal(ocPaymentMethodBalance.currency);
+    });/** END OF "check if collective balance for Admin Member matches sum of transactions amounts" */
+
+    it('Collective 1 fails to pay Collective 2 because it is not logged in', async () => {
+      //finding opencollective payment method for Collective 1
+      openCollectivePaymentMethod = await models.PaymentMethod.findOne({ where: {type: 'collective', CollectiveId: collective1.id}});
+
+      // get Balance given the created user
+      const ocPaymentMethodBalance = await openCollectivePaymentMethod.getBalanceForUser(user1);
+
+      // Setting up order
+      const order = {
+        fromCollective: { id: collective1.id },
+        collective: { id: collective2.id },
+        paymentMethod: { uuid: openCollectivePaymentMethod.uuid },
+        totalAmount: ocPaymentMethodBalance.amount,
+      }
+      // Executing queries
+      const res = await utils.graphqlQuery(createOrderQuery, { order });
+
+      // Then there should be Errors
+      expect(res.errors).to.exist;
+
+    });/** END OF "Collective 1 fails to pay Collective 2 because it is not logged in" */
+
+    it('Collective 1 fails to pay Collective 2 because user is not Admin', async () => {
```

Comment:
Can probably be consolidated with the previous test ("Non admin users can't use the payment method of a collective")